### PR TITLE
Support DISTINCT

### PR DESCRIPTION
**Pull request by xuebinsu**
This patch adds support for DISTINCT, both SELECT DISTINCT and distinct-qualified aggregation like count(DISTINCT). Specifically, distinct() are added to both `Table` and `AggregateFunction` as methods, respectively.

This patch also fixes error when assign() is empty after group_by(), changing it to truly return a `Table`, as specified in the signature.

### CODE REVIEW

File: greenplumpython/func.py [hunk 1]
```diff
@@ -44,23 +46,27 @@ def __init__(
...
         self._args = args
         self._group_by = group_by
+        self._distinct = distinct
 
     def bind(
         self,
         group_by: Optional[TableGroupingSets] = None,
         table: Optional[Table] = None,
         db: Optional[Database] = None,
+        distinct: Optional[bool] = None,
```
Comment (yihong0618) [inline greenplumpython/func.py:56]: do we need add test for .bind(..., distinct=True) or something

File: greenplumpython/func.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import inspect
 import re
 import textwrap
+from dis import dis
```
Comment (yihong0618) [inline greenplumpython/func.py:8]: no need to add this IMO

**Review by Quasimewdo — APPROVED** [review]
LGTM

**Review by yihong0618 — APPROVED** [review]
LGTM

**Pull request merged**